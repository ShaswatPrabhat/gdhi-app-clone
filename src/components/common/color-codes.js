export default {
  getColorCodes() {
    // Update variables if changed.
    return [
      { score: "5", color: "#01C975", description: "Most Developed" },
      { score: "4", color: "#80E1CC", description: "" },
      { score: "3", color: "#FFE180", description: "" },
      { score: "2", color: "#FFCA82", description: "" },
      { score: "1", color: "#FCAB9C", description: "Least Developed" },
      { score: "NA", color: "#6C757D" },
    ];
  },
};
