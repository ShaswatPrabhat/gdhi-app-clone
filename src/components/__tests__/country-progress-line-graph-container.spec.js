import { createLocalVue, shallowMount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import { describe, beforeEach, it, expect, vi } from "vitest";
import CountryProgressLineGraphContainer from "../graphs/country-progress-line-graph/country-progress-line-graph-container.vue";
import { EventBus } from "../common/event-bus";
import axios from "axios";
import VueRouter from "vue-router";

const eventBusOnSpy = vi.spyOn(EventBus, "$on");
describe("Country Progress Line Graph Container", () => {
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();
  let wrapper;
  const yearsResponse = {
    years: ["2023", "2022", "2021"],
    defaultYear: "2023",
  };
  const yearOnYearData = {
    currentYear: "2023",
    defaultYear: "2022",
    yearOnYearData: [
      {
        year: "2023",
        data: {
          country: {
            countryId: "IND",
            countryName: "India",
            countryAlpha2Code: "IN",
            categories: [],
            countryPhase: 3,
            updatedDate: "",
          },
          average: {},
        },
      },
    ],
  };
  const axiosGetSpy = vi.spyOn(axios, "get");
  axiosGetSpy.mockImplementation(async (url) => {
    if (url.includes("year_on_year")) {
      return new Promise((resolve) => resolve({ data: yearOnYearData }));
    } else if (url.includes("distinct_year")) {
      return new Promise((resolve) => resolve({ data: yearsResponse }));
    }
  });
  window.appProperties = {
    getCategoryFilter: function () {
      return "";
    },
    getPhaseFilter: function () {
      return "";
    },
  };
  beforeEach(() => {
    wrapper = shallowMount(CountryProgressLineGraphContainer, {
      propsData: {
        countryName: "India",
        locale: "en",
      },
      localVue,
      router,
    });
  });

  it("should render country progress line graph container", async () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should fetch yearOnYearData on mount", async () => {
    wrapper.vm.getYearOnYearData("IND");
    await flushPromises();
    expect(wrapper.vm.yearOnYearData).to.deep.equal(yearOnYearData);
  });

  it("should fetch distinct years on mount", async () => {
    await flushPromises();
    expect(wrapper.vm.years).to.deep.equal(yearsResponse.years);
  });

  it("should register event when line graph container is mounted", async () => {
    expect(eventBusOnSpy.mock.calls[0][0]).to.equal("indicator:filtered");
  });
});
