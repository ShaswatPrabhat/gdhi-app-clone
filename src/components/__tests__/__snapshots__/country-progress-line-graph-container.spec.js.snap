// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Country Progress Line Graph Container > should render country progress line graph container 1`] = `
<div
  class="container"
  data-v-b2657d67=""
>
  <div
    class="indicator-filter-container"
    data-v-b2657d67=""
  >
    <indicatorfilter-stub
      data-v-b2657d67=""
      title="countryProfile.countryProgressLineChart.indicatorGroupingDescription"
    />
  </div>
  <div
    class="spinner"
    data-v-b2657d67=""
  >
    <i
      class="fa fa-spinner fa-pulse fa-3x fa-fw"
      data-v-b2657d67=""
    />
  </div>
</div>
`;
